Test failure with pandas 1.1

```
_______________________________ test_get_dask_array[0.5] _______________________________

self = <[AttributeError("'Categorical' object has no attribute '_codes'") raised in repr()] Categorical object at 0x7f2b5ec9c400>
values = array([<xarray.DataArray 'train_pulse' ()>
array(10000, dtype=uint64)
Coordinates:
    train_pulse  uint64 10000,
    ...array.DataArray 'train_pulse' ()>
array(10479, dtype=uint64)
Coordinates:
    train_pulse  uint64 10479], dtype=object)
categories = None, ordered = False
dtype = CategoricalDtype(categories=None, ordered=False), fastpath = False

    def __init__(
        self, values, categories=None, ordered=None, dtype=None, fastpath=False
    ):
    
        dtype = CategoricalDtype._from_values_or_dtype(
            values, categories, ordered, dtype
        )
        # At this point, dtype is always a CategoricalDtype, but
        # we may have dtype.categories be None, and we need to
        # infer categories in a factorization step further below
    
        if fastpath:
            self._codes = coerce_indexer_dtype(values, dtype.categories)
            self._dtype = self._dtype.update_dtype(dtype)
            return
    
        # null_mask indicates missing values we want to exclude from inference.
        # This means: only missing values in list-likes (not arrays/ndframes).
        null_mask = np.array(False)
    
        # sanitize input
        if is_categorical_dtype(values):
            if dtype.categories is None:
                dtype = CategoricalDtype(values.categories, dtype.ordered)
        elif not isinstance(values, (ABCIndexClass, ABCSeries)):
            # sanitize_array coerces np.nan to a string under certain versions
            # of numpy
            values = maybe_infer_to_datetimelike(values, convert_dates=True)
            if not isinstance(values, np.ndarray):
                values = com.convert_to_list_like(values)
    
                # By convention, empty lists result in object dtype:
                sanitize_dtype = np.dtype("O") if len(values) == 0 else None
                null_mask = isna(values)
                if null_mask.any():
                    values = [values[idx] for idx in np.where(~null_mask)[0]]
                values = sanitize_array(values, None, dtype=sanitize_dtype)
    
        if dtype.categories is None:
            try:
>               codes, categories = factorize(values, sort=True)

../../.local/lib/python3.8/site-packages/pandas/core/arrays/categorical.py:343: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

values = array([<xarray.DataArray 'train_pulse' ()>
array(10000, dtype=uint64)
Coordinates:
    train_pulse  uint64 10000,
    ...array.DataArray 'train_pulse' ()>
array(10479, dtype=uint64)
Coordinates:
    train_pulse  uint64 10479], dtype=object)
sort = True, na_sentinel = -1, size_hint = None, dropna = True

    @doc(
        values=dedent(
            """\
        values : sequence
            A 1-D sequence. Sequences that aren't pandas objects are
            coerced to ndarrays before factorization.
        """
        ),
        sort=dedent(
            """\
        sort : bool, default False
            Sort `uniques` and shuffle `codes` to maintain the
            relationship.
        """
        ),
        size_hint=dedent(
            """\
        size_hint : int, optional
            Hint to the hashtable sizer.
        """
        ),
    )
    def factorize(
        values,
        sort: bool = False,
        na_sentinel: int = -1,
        size_hint: Optional[int] = None,
        dropna: bool = True,
    ) -> Tuple[np.ndarray, Union[np.ndarray, ABCIndex]]:
        """
        Encode the object as an enumerated type or categorical variable.
    
        This method is useful for obtaining a numeric representation of an
        array when all that matters is identifying distinct values. `factorize`
        is available as both a top-level function :func:`pandas.factorize`,
        and as a method :meth:`Series.factorize` and :meth:`Index.factorize`.
    
        Parameters
        ----------
        {values}{sort}
        na_sentinel : int, default -1
            Value to mark "not found".
        {size_hint}\
    
        Returns
        -------
        codes : ndarray
            An integer ndarray that's an indexer into `uniques`.
            ``uniques.take(codes)`` will have the same values as `values`.
        uniques : ndarray, Index, or Categorical
            The unique valid values. When `values` is Categorical, `uniques`
            is a Categorical. When `values` is some other pandas object, an
            `Index` is returned. Otherwise, a 1-D ndarray is returned.
    
            .. note ::
    
               Even if there's a missing value in `values`, `uniques` will
               *not* contain an entry for it.
    
        See Also
        --------
        cut : Discretize continuous-valued array.
        unique : Find the unique value in an array.
    
        Examples
        --------
        These examples all show factorize as a top-level method like
        ``pd.factorize(values)``. The results are identical for methods like
        :meth:`Series.factorize`.
    
        >>> codes, uniques = pd.factorize(['b', 'b', 'a', 'c', 'b'])
        >>> codes
        array([0, 0, 1, 2, 0]...)
        >>> uniques
        array(['b', 'a', 'c'], dtype=object)
    
        With ``sort=True``, the `uniques` will be sorted, and `codes` will be
        shuffled so that the relationship is the maintained.
    
        >>> codes, uniques = pd.factorize(['b', 'b', 'a', 'c', 'b'], sort=True)
        >>> codes
        array([1, 1, 0, 2, 1]...)
        >>> uniques
        array(['a', 'b', 'c'], dtype=object)
    
        Missing values are indicated in `codes` with `na_sentinel`
        (``-1`` by default). Note that missing values are never
        included in `uniques`.
    
        >>> codes, uniques = pd.factorize(['b', None, 'a', 'c', 'b'])
        >>> codes
        array([ 0, -1,  1,  2,  0]...)
        >>> uniques
        array(['b', 'a', 'c'], dtype=object)
    
        Thus far, we've only factorized lists (which are internally coerced to
        NumPy arrays). When factorizing pandas objects, the type of `uniques`
        will differ. For Categoricals, a `Categorical` is returned.
    
        >>> cat = pd.Categorical(['a', 'a', 'c'], categories=['a', 'b', 'c'])
        >>> codes, uniques = pd.factorize(cat)
        >>> codes
        array([0, 0, 1]...)
        >>> uniques
        ['a', 'c']
        Categories (3, object): ['a', 'b', 'c']
    
        Notice that ``'b'`` is in ``uniques.categories``, despite not being
        present in ``cat.values``.
    
        For all other pandas objects, an Index of the appropriate type is
        returned.
    
        >>> cat = pd.Series(['a', 'a', 'c'])
        >>> codes, uniques = pd.factorize(cat)
        >>> codes
        array([0, 0, 1]...)
        >>> uniques
        Index(['a', 'c'], dtype='object')
        """
        # Implementation notes: This method is responsible for 3 things
        # 1.) coercing data to array-like (ndarray, Index, extension array)
        # 2.) factorizing codes and uniques
        # 3.) Maybe boxing the uniques in an Index
        #
        # Step 2 is dispatched to extension types (like Categorical). They are
        # responsible only for factorization. All data coercion, sorting and boxing
        # should happen here.
    
        values = _ensure_arraylike(values)
        original = values
    
        if is_extension_array_dtype(values.dtype):
            values = extract_array(values)
            codes, uniques = values.factorize(na_sentinel=na_sentinel)
            dtype = original.dtype
        else:
            values, dtype = _ensure_data(values)
    
            if original.dtype.kind in ["m", "M"]:
                na_value = na_value_for_dtype(original.dtype)
            else:
                na_value = None
    
>           codes, uniques = _factorize_array(
                values, na_sentinel=na_sentinel, size_hint=size_hint, na_value=na_value
            )

../../.local/lib/python3.8/site-packages/pandas/core/algorithms.py:647: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

values = array([<xarray.DataArray 'train_pulse' ()>
array(10000, dtype=uint64)
Coordinates:
    train_pulse  uint64 10000,
    ...array.DataArray 'train_pulse' ()>
array(10479, dtype=uint64)
Coordinates:
    train_pulse  uint64 10479], dtype=object)
na_sentinel = -1, size_hint = None, na_value = None, mask = None

    def _factorize_array(
        values, na_sentinel: int = -1, size_hint=None, na_value=None, mask=None,
    ) -> Tuple[np.ndarray, np.ndarray]:
        """
        Factorize an array-like to codes and uniques.
    
        This doesn't do any coercion of types or unboxing before factorization.
    
        Parameters
        ----------
        values : ndarray
        na_sentinel : int, default -1
        size_hint : int, optional
            Passed through to the hashtable's 'get_labels' method
        na_value : object, optional
            A value in `values` to consider missing. Note: only use this
            parameter when you know that you don't have any values pandas would
            consider missing in the array (NaN for float data, iNaT for
            datetimes, etc.).
        mask : ndarray[bool], optional
            If not None, the mask is used as indicator for missing values
            (True = missing, False = valid) instead of `na_value` or
            condition "val != val".
    
        Returns
        -------
        codes : ndarray
        uniques : ndarray
        """
        hash_klass, values = _get_data_algo(values)
    
        table = hash_klass(size_hint or len(values))
>       uniques, codes = table.factorize(
            values, na_sentinel=na_sentinel, na_value=na_value, mask=mask
        )

../../.local/lib/python3.8/site-packages/pandas/core/algorithms.py:495: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???

pandas/_libs/hashtable_class_helper.pxi:1873: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   TypeError: unhashable type: 'DataArray'

pandas/_libs/hashtable_class_helper.pxi:1787: TypeError

During handling of the above exception, another exception occurred:

mock_fxe_raw_run = '/tmp/tmprf7bqp77'

    def test_get_dask_array(mock_fxe_raw_run):
        run = RunDirectory(mock_fxe_raw_run)
        det = LPD1M(run)
>       arr = det.get_dask_array('image.data')

extra_data/tests/test_components.py:115: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
extra_data/components.py:392: in get_dask_array
    mod_arr.coords['train_pulse'] = pd.MultiIndex.from_arrays(
../../.local/lib/python3.8/site-packages/pandas/core/indexes/multi.py:439: in from_arrays
    codes, levels = factorize_from_iterables(arrays)
../../.local/lib/python3.8/site-packages/pandas/core/arrays/categorical.py:2725: in factorize_from_iterables
    return map(list, zip(*(factorize_from_iterable(it) for it in iterables)))
../../.local/lib/python3.8/site-packages/pandas/core/arrays/categorical.py:2725: in <genexpr>
    return map(list, zip(*(factorize_from_iterable(it) for it in iterables)))
../../.local/lib/python3.8/site-packages/pandas/core/arrays/categorical.py:2697: in factorize_from_iterable
    cat = Categorical(values, ordered=False)
../../.local/lib/python3.8/site-packages/pandas/core/arrays/categorical.py:345: in __init__
    codes, categories = factorize(values, sort=False)
../../.local/lib/python3.8/site-packages/pandas/core/algorithms.py:647: in factorize
    codes, uniques = _factorize_array(
../../.local/lib/python3.8/site-packages/pandas/core/algorithms.py:495: in _factorize_array
    uniques, codes = table.factorize(
pandas/_libs/hashtable_class_helper.pxi:1873: in pandas._libs.hashtable.PyObjectHashTable.factorize
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   TypeError: unhashable type: 'DataArray'

pandas/_libs/hashtable_class_helper.pxi:1787: TypeError
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -10,7 +10,7 @@ env:
 
 install:
   - make install
-  - pip install -U pytest  # pytest-cov needs a newer pytest
+  - pip install -U 'pytest<6'  # pytest-cov needs a newer pytest, but nbval currently fails with pytest 6
   - pip install codecov
 
 script:

--- a/extra_data/components.py
+++ b/extra_data/components.py
@@ -390,7 +390,7 @@ class MPxDetectorBase:
                 mod_arr = mod_arr.rename({'trainId': 'train_pulse'})
 
                 mod_arr.coords['train_pulse'] = pd.MultiIndex.from_arrays(
-                    [mod_arr.coords['train_pulse'], inner_ix],
+                    [mod_arr.coords['train_pulse'].values, inner_ix.values],
                     names=['trainId', subtrain_index]
                 )
 

